pkg/provider/doppler/client.go

```diff
@@ -70,16 +73,38 @@ type SecretsClientInterface interface {
 }
 
 func (c *Client) setAuth(ctx context.Context) error {
-	token, err := resolvers.SecretKeyRef(
-		ctx,
-		c.kube,
-		c.storeKind,
-		c.namespace,
-		&c.store.Auth.SecretRef.DopplerToken)
-	if err != nil {
-		return err
+	if c.store.Auth.SecretRef != nil {
+		token, err := resolvers.SecretKeyRef(
+			ctx,
+			c.kube,
+			c.storeKind,
+			c.namespace,
+			&c.store.Auth.SecretRef.DopplerToken)
+		if err != nil {
+			return err
+		}
+		c.dopplerToken = token
+	} else if c.store.Auth.OIDCConfig != nil {
+		token, err := c.oidcManager.GetToken(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get OIDC token: %w", err)
+		}
+		c.dopplerToken = token
+	}
+	return nil
+}
+
+func (c *Client) refreshAuthIfNeeded(ctx context.Context) error {
+	if c.store != nil && c.store.Auth != nil && c.store.Auth.OIDCConfig != nil && c.oidcManager != nil {
+		token, err := c.oidcManager.GetToken(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to refresh OIDC token: %w", err)
+		}
+		c.dopplerToken = token
+		if doppler, ok := c.doppler.(*dClient.DopplerClient); ok {
+			doppler.DopplerToken = token
```

Comment:
check/question: Why is this part necessary? It doesn't look like we were doing this before.